### PR TITLE
Add support for configmap in checkpointer

### LINKED ISSUE #124: Support check pointing configmaps
Chatted with @derekparker about this.

Currently bootkube checkpoints secrets for the bootstrapped API server. This lets the API server refer to secrets in it's manifest (e.g. for TLS assets, config flags, etc). It doesn't support configmaps, which might be more appropriate for non-secret configuration. In my case, this would be a policy file.

### PR DESCRIPTION
**Pull request by dhawal55**
Fixes #124

### CODE REVIEW

**Comment by aaronlevy** [comment]
ok to test

**Comment by pbx0** [comment]
OK to test

**Comment by pbx0** [comment]
OK to test

**Comment by pbx0** [comment]
OK to test

**Comment by pbx0** [comment]
test this please

**Comment by pbx0** [comment]
ok to test

**Comment by pbx0** [comment]
rktbot run tests

**Comment by pbx0** [comment]
ok to test

**Comment by pbx0** [comment]
rktbot run etcd tests

**Comment by pbx0** [comment]
rktbot run tests
rktbot run etcd tests
rktbot run checkpointer tests

**Comment by pbx0** [comment]
rktbot run etcd tests

**Comment by pbx0** [comment]
rktbot run checkpointer tests

**Comment by pbx0** [comment]
rktbot run checkpointer tests

**Comment by pbx0** [comment]
rktbot run checkpointer tests

**Comment by pbx0** [comment]
rktbot run checkpointer tests

**Comment by pbx0** [comment]
Sorry for the noise, just testing out bot functionality.

**Comment by dhawal55** [comment]
Can someone publish details of the failed build (bootkube-etcd-dev) and let me know how i can fix it?

**Comment by aaronlevy** [comment]
@dhawal55 the PR tests are a bit in a state of flux - and there are some flakes we're still working through (and not currently able to expose the logs publicly - but it's on roadmap). Sorry for the delay on this (and thank you for the contribution!). I'll try and take a look at this/test early next week.

File: cmd/checkpoint/main.go [hunk 1]
```diff
@@ -213,6 +214,14 @@ func createCheckpointsForValidParents(client clientset.Interface, pods map[strin
 			glog.Errorf("Failed to checkpoint secrets for pod %s: %v", id, err)
 			continue
 		}
+		cp, err = checkpointConfigMapVolumes(client, cp)
+		if err != nil {
+			//TODO(aaron): This can end up spamming logs at times when api-server is unavailable. To reduce spam
+			//             we could only log error if api-server can't be contacted and existing secret doesn't exist.
```
Comment (aaronlevy) [inline cmd/checkpoint/main.go]: s/existing secret/existing configmap/

**Comment by aaronlevy** [comment]
Overall this looks good, thank you for contributing this!

I'll run some tests against it, but assuming those pass - we should be able to get this merged soon.

**Comment by aaronlevy** [comment]
rktbot run tests

**Comment by aaronlevy** [comment]
The PR tests might actually be down right now (due to S3 outages).

**Comment by pbx0** [comment]
rktbot run tests

File: cmd/checkpoint/main.go (hunk 1 above)
Comment (dhawal55) [inline cmd/checkpoint/main.go]: ah, good catch, it's a copy paste mistake. I will correct it

**Comment by aaronlevy** [comment]
@dhawal55 I'm going to do some manual sanity checks on this now - would you also mind squashing your commits?

**Comment by aaronlevy** [comment]
@dhawal55 also looks like there are some conflicts that need to be resolved

**Comment by pbx0** [comment]
rktbot run checkpointer tests

**Comment by pbx0** [comment]
rktbot run checkpointer tests

**Comment by dhawal55** [comment]
Squashed the commits and fixed merge issues

**Comment by aaronlevy** [comment]
rktbot run checkpointer tests

**Comment by aaronlevy** [comment]
I manually tested this and it worked. The checkpointer-dev test I believe has an unrelated issue.

@pbx0 when you get a chance can you look into the failed test. It looks like re-running the etcd-scale test the second time  failed.

**Pull request merged**